### PR TITLE
CTest Fixes

### LINKED ISSUE #9948: CTest fails on RHEL7
#### Description of problem:
Following (CTest) tests fail on RHEL-7.9:
```
The following tests FAILED:
	 60 - test-compare_ds (Failed)
	 64 - test-controleval-directory (Failed)
	 65 - test-controleval-onefile (Failed)
	 66 - test-controleval-json (Failed)
Errors while running CTest
gmake[1]: *** [test] Error 8
```

#### SCAP Security Guide Version:
master (tested on commit e7502af)

#### Operating System Version:
RHEL-7.9

#### Steps to Reproduce:
1. Run CTest

#### Actual Results:
```
 60/102 Test  #60: test-compare_ds ....................................................***Failed    0.06 sec
Traceback (most recent call last):
  File "/root/rpmbuild/BUILD/scap-security-guide-0.1.66/utils/compare_ds.py", line 6, in <module>
    from ssg.content_diff import StandardContentDiffer, StigContentDiffer
  File "/root/rpmbuild/BUILD/scap-security-guide-0.1.66/ssg/content_diff.py", line 7, in <module>
    import xml.etree.ElementTree as ET
ImportError: No module named etree.ElementTree
...
 64/102 Test  #64: test-controleval-directory .........................................***Failed    0.02 sec
  File "/root/rpmbuild/BUILD/scap-security-guide-0.1.66/utils/controleval.py", line 31
    raise RuntimeError(msg) from e
                               ^
SyntaxError: invalid syntax

        Start  65: test-controleval-onefile
 65/102 Test  #65: test-controleval-onefile ...........................................***Failed    0.02 sec
  File "/root/rpmbuild/BUILD/scap-security-guide-0.1.66/utils/controleval.py", line 31
    raise RuntimeError(msg) from e
                               ^
SyntaxError: invalid syntax

        Start  66: test-controleval-json
 66/102 Test  #66: test-controleval-json ..............................................***Failed    0.01 sec
  File "/root/rpmbuild/BUILD/scap-security-guide-0.1.66/utils/controleval.py", line 31
    raise RuntimeError(msg) from e
                               ^
SyntaxError: invalid syntax
```

#### Expected Results:
no failing tests

### LINKED ISSUE #9958: Some RHEL9 specific tests run even when RHEL9 product is not built
#### Description of problem:

The following tests seem to be applicable only for RHEL9:
- srg-export-rhel9
- srg-diff-rhel9
However, they seem to run even when rhel9 product is not built. They fail, poluting test results.

#### SCAP Security Guide Version:

master, currently at 4300b3e4b7ecec9fd1d65c0d772c1fc39503cfa6  

#### Operating System Version:

Fedora 36

#### Steps to Reproduce:

1. ./build_product rhel8
2. cd build
3. ctest -L quick

#### Actual Results:

         41 - srg-export-rhel9 (Failed)
         42 - srg-diff-rhel9 (Not Run)

#### Expected Results:

Ideally no failures if the content is correct. These tests should definitely not fail, tehy should be skipped.

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

This PR fixes a couple of issues with the CTest test suite.

#### Rationale:

* Fixes #9948
* Fixes #9958
#### Review Hints:
* Make sure to run the tests when you have not built `rhel9`.

### CODE REVIEW

**Review by jan-cerny — APPROVED** [review]
I have checked in a RHEL 7.9 VM that with this PR the tests mentioned in #9948 stopped to be executed. I also checked on F37 that the tests srg-export-rhel9 and srg-diff-rhel9 aren't executed when rhel8 content is built.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -160,7 +160,7 @@ if (PY_MYPY)
      )
 endif()
 
-if (PY_OPENPYXL AND PY_PANDAS AND PYTHON3_EXECUTABLE)
+if (PYTHON_VERSION_MAJOR GREATER 2 AND PY_OPENPYXL AND PY_PANDAS AND SRG_PRODUCT_RHEL9)
```
Comment (ggbecker) [inline tests/CMakeLists.txt:163]: isn't it `SSG_PRODUCT_RHEL9` instead of `SRG...`?

Comment (jan-cerny) [reply tests/CMakeLists.txt:163]: great, thanks for finding this, it should be SSG_PRODUCT_RHEL9

@Mab879 PTAL

Comment (Mab879) [reply tests/CMakeLists.txt:163]: https://github.com/ComplianceAsCode/content/pull/9964 should fix that.

**Pull request merged**